libs/api/domains/air-discount-scheme/src/lib/discount/discount.resolver.ts

```diff
@@ -11,16 +11,37 @@ import {
 import { Audit } from '@island.is/nest/audit'
 import { DiscountService } from './discount.service'
 import { Discount } from '../models/discount.model'
+import { FlightLeg } from '../models/flightLeg.model'
+import { User as Relation } from '../models/user.model'
+import type { FlightLeg as TFlightLeg } from '@island.is/clients/air-discount-scheme'
 
 @UseGuards(IdsUserGuard, ScopesGuard)
 @Scopes(ApiScope.internal)
 @Audit({ namespace: '@island.is/air-discount-scheme' })
-@Resolver()
+@Resolver(() => Discount)
 export class DiscountResolver {
   constructor(private discountService: DiscountService) {}
 
   @Query(() => [Discount], { name: 'airDiscountSchemeDiscounts' })
-  async getDiscount(@CurrentUser() user: User) {
+  async getDiscount(@CurrentUser() user: User): Promise<Discount[]> {
     return this.discountService.getCurrentDiscounts(user)
   }
+
+  @ResolveField('flightLegs', () => [FlightLeg])
+  async resolveFlightLegs(
+    @CurrentUser() user: User,
+    @Parent() relation: Relation,
+  ): Promise<TFlightLeg[]> {
+    // AirDiscountSChemeDiscounts yields discounts for the authenticated user
+    // and any eligible wards (relations).
+    return this.discountService.getFlightLegsByNationalId(
+      user,
+      relation.nationalId,
+    )
```

Comment:
I have concerns about this endpoint which can query flight legs for any arbitrary national id. I know the service is inside our VPN and the endpoint is only exposed like this for "child" national ids. But it still unnecessarily opens us up for data leaks in case of VPN breaches.

What's the usage pattern like? Is this heavy? Does the client sometimes request this data and sometimes not? Does the client fetch this data one at a time or for all national ids at the same time?

Depending on the answers, we either might want to add this data to the original discount request and/or update the controller to check if CurrentUser has access to the specified `nationalId`.

Reply:
I'm opting for including this in the response instead. Good call.

Reply:
Ok, I included this in the discount response.

There may be a case where flights and flight-legs are present in the system but a person isn't eligible for discounts so I added another endpoint, tied to the user, which only displays that person's flight legs, which is consistent with the current implementation of Air Discount Scheme.